src/main/java/digital/slovensko/autogram/core/Autogram.java

```diff
@@ -31,7 +33,23 @@ public Autogram(UI ui, DriverDetector driverDetector) {
     }
 
     public void sign(SigningJob job) {
-        ui.onUIThreadDo(() -> ui.startSigning(job, this));
+        ui.onUIThreadDo(()
+        -> ui.startSigning(job, this));
+    }
+
+    public void checkAndValidateSignatures(SigningJob job) {
+        ui.onWorkThreadDo(() -> checkSignatures(job));
+
+        if (!SignatureValidator.hasSignatures(job.getDocument()))
+            return;
+
+        var reports = SignatureValidator.getInstance().getSignatureValidationReport(job.getDocument());
```

Comment:
Nedava zmysel, aby nas validator dostal job a vratil uz ten ValidationReports? (Wrapper by som dropol, ved to je nas namespace)